src/bthome_ble/parser.py

```diff
@@ -555,17 +560,30 @@ def _decrypt_bthome(
         else:
             uuid = b"\xd2\xfc" + bytes([adv_info])
         encrypted_payload = data[:-8]
-        count_id = data[-8:-4]
+        last_encryption_counter = self.encryption_counter
+        self.encryption_counter = data[-8:-4]
         mic = data[-4:]
 
-        # nonce: mac [6], uuid16 [2 (v1) or 3 (v2)], count_id [4]
-        nonce = b"".join([bthome_mac, uuid, count_id])
+        # nonce: mac [6], uuid16 [2 (v1) or 3 (v2)], encryption_counter [4]
+        nonce = b"".join([bthome_mac, uuid, self.encryption_counter])
 
         associated_data = None
         if sw_version == 1:
             associated_data = b"\x11"
 
         assert self.cipher is not None  # nosec
+
+        # verify that the encryption counter is increasing compared the previous value
+        if (
+            self.encryption_counter < last_encryption_counter
+            and last_encryption_counter < b"\xF0\xFF\xFF\xFF"
```

Comment:
I think these need to be converted to int values first.

```
>>> b'\xff\x00\x00\x00' < b'\x00\xff\xff\x00'
False
>>> int.from_bytes(b'\xff\x00\x00\x00', "little", signed=False) < int.from_bytes(b'\x00\xff\xff\x00', "little", signed=False)
True
```

Reply:
Fixed